### ISSUE TITLE
tx-selector shouldn't check toEthAddr in transferToIdx txs

### ISSUE THREAD

**Issue by arnaubennassar** [hermeznetwork/hermez-node#749]
## Summary of Bug

`tx-selector` checks that ethereum address matches with toIdx referenced account when doing a transfer to idx. However this is doable in terms of protocol (not checked anywhere).

### Expected Behavior

`tx-selector` should select transactions with valid toIdx even if the toEthAddr is set to something random

### Steps to Reproduce

Send a valid tx toIdx, with some random value in the toEthAddr

### Additional Information:

An example of such a tx

**Comment by arnaubennassar** [comment]
Some extra details:

- Changes introduced on this PR #708 will block "weird" transactions, so they won't hit the pool
- `tx-selector` is not consistent with the [protocol](https://docs.hermez.io/#/developers/protocol/hermez-protocol/protocol?id=transfer), as the value of `ToEthAddr` and `ToBJJ` should be ignored when `ToIdx > 255`:
    - https://github.com/hermeznetwork/hermez-node/blob/develop/txselector/txselector.go#L502
    - https://github.com/hermeznetwork/hermez-node/blob/develop/txselector/txselector.go#L519

In other words, we have a bug in the `tx-selector` that should never happen due to a "extra security check" on the `api`.
In the future we may stop performing those sanity checks ( #751 ), so we must fix this bug.
